### DISCUSSION TITLE
Include some python files in esp32 image without rebuilding

### DISCUSSION THREAD

**Discussion by RobertoD91** [micropython/micropython#16145] [Using MicroPython]
### Short version:
Hi, Is there a way to add Python files to an ESP32 MicroPython image without having to rebuild it?

### Long version:
Hi,
I'm using MicroPython on an ESP32 in a high school course. The school recently purchased this robot: [Q-scout](https://www.robobloq.com/product/Q-scout), but the included IDE is unusable.
Fortunately it's esp32 based and use micropython, so I managed to extract the Python scripts provided by the company for controlling the proprietary motors and sensors, and they work flawlessly on the latest MicroPython version.

For development, I'm using Viper-IDE and mpremote for coding and debugging, and they’re working really well.

The only thing I'm missing is a pre-configured ESP32 image that I can give to the students to quickly reset the robot in case of issues (since another course is using a different firmware on the same esp32 and we have a lot of esp32 to flash...).

I know it's possible to rebuild MicroPython to create a custom image with included files, but I wanted to ask if there's a way to simply add files to the official MicroPython image without having to rebuild it.

Thanks in advance!

**Comment by jonnor** [comment]
Since a few weeks back Viper IDE supports examples that can be one-click installed by clicking a link on a webpage. Or you can paste the URL inside the IDE under Install from Link. An example can include multiple files (mip dependencies, Python files, .mpy C modules, data files etc). So that might be a workable mechanism?
See https://github.com/vshymanskyy/ViperIDE/issues/42

**Reply by jonnor** [reply]
Here is an example of it in use
https://github.com/emlearn/emlearn-micropython/tree/master/examples/xor_trees#running-on-device-viper-ide
And the package.json that makes it work
https://github.com/emlearn/emlearn-micropython/blob/master/examples/xor_trees/package.json

**Reply by RobertoD91** [reply]
Yes, thanks! This looks great. I’ll test it asap

**Comment by Josverl** [comment]
Another option is to use vsf_merge to merge a Micropython image with a folder structure containing the files.

The resulting image can be flashed in one go, and will contain both


https://github.com/Josverl/vfs_merge

**Reply by RobertoD91** [reply]
Thank you! I was looking for something like this; I’ll test it asap
